src/App.js

```diff
@@ -1,9 +1,59 @@
-function App() {
+import styled from 'styled-components/macro'
+import { useEffect, useState, useRef } from 'react'
+import Header from './components/Header'
+import MainButton from './components/MainButton'
+import audioUrl from './assets/audio/music/Uno dos tres.mp3'
+
+export default function App() {
+  const [appState, setAppState] = useState('default')
+  let songRef = useRef(null)
+
+  useEffect(() => {
+    songRef.current = new Audio(audioUrl)
+  }, [])
+
+  useEffect(() => {
+    if (appState === 'default' || appState === 'paused') {
+      songRef.current.pause()
+    } else {
+      songRef.current.play()
+    }
+  }, [appState])
+
+  function toogleSongPlay() {
+    setAppState(appState === 'playing' ? 'paused' : 'playing')
+  }
+
+  function stopPlaying() {
+    songRef.current.currentTime = 0
+    setAppState('default')
+  }
+
   return (
-    <div className="App">
-     
-    </div>
-  );
+    <AppStyled>
+      <Header appState={appState} handleClick={stopPlaying} />
+      <main></main>
+      <footer>
+        <MainButton appState={appState} handleClick={toogleSongPlay} />
+      </footer>
+    </AppStyled>
+  )
 }
 
-export default App;
+const AppStyled = styled.div`
+  display: grid;
+  grid-template-rows: 80px auto 80px;
+  height: 100%;
+  color: var(--text-color);
+  background-color: var(--primery-color);
```

Comment:
```suggestion
  background-color: var(--primary-color);
```
little typo in primary :)

Reply:
You got me again...